packages/lit-node-client-nodejs/src/lib/lit-node-client-nodejs.ts

```diff
@@ -2472,75 +1999,49 @@ export class LitNodeClientNodeJs
     }
 
     let siweMessage;
 
-    if (params?.resourceAbilityRequests) {
-      const resources = params.resourceAbilityRequests.map((r) => r.resource);
-
-      const recapObject =
-        await this.generateSessionCapabilityObjectWithWildcards(resources);
-
-      params.resourceAbilityRequests.forEach((r) => {
-        recapObject.addCapabilityForResource(r.resource, r.ability);
-
-        const verified = recapObject.verifyCapabilitiesForResource(
-          r.resource,
-          r.ability
-        );
-
-        if (!verified) {
-          throw new Error('Failed to verify capabilities for resource');
-        }
-      });
+    const siweParams = {
+      domain: params?.domain || globalThis.location?.host || 'litprotocol.com',
+      walletAddress: pkpEthAddress,
+      statement: siwe_statement,
+      uri: sessionKeyUri,
+      version: '1',
+      chainId: params.chainId ?? 1,
+      expiration: _expiration,
+      nonce: this.latestBlockhash!,
+    };
 
-      // regular siwe
-      siweMessage = new siwe.SiweMessage({
-        domain:
-          params?.domain || globalThis.location?.host || 'litprotocol.com',
-        address: pkpEthAddress,
-        statement: siwe_statement,
-        uri: sessionKeyUri,
-        version: '1',
-        chainId: params.chainId ?? 1,
-        expirationTime: _expiration,
-        resources: params.resources,
-        nonce: this.latestBlockhash!,
+    if (params.resourceAbilityRequests) {
+      siweMessage = await createSiweMessageWithRecaps({
+        ...siweParams,
+        resources: params.resourceAbilityRequests,
+        litNodeClient: this,
       });
-
-      siweMessage = recapObject.addToSiweMessage(siweMessage);
     } else {
-      // lit-siwe (NOT regular siwe)
-      siweMessage = new siwe.SiweMessage({
-        domain:
-          params?.domain || globalThis.location?.host || 'litprotocol.com',
-        address: pkpEthAddress,
-        statement: siwe_statement,
-        uri: sessionKeyUri,
-        version: '1',
-        chainId: params.chainId ?? 1,
-        expirationTime: _expiration,
-        resources: params.resources,
-        nonce: this.latestBlockhash!,
-      });
+      siweMessage = await createSiweMessage(siweParams);
     }
 
-    const siweMessageStr: string = (
-      siweMessage as siwe.SiweMessage
-    ).prepareMessage();
-
     // ========== Get Node Promises ==========
     // -- fetch shares from nodes
-    const body = {
+    const body: JsonSignSessionKeyRequestV1 = {
       sessionKey: sessionKeyUri,
       authMethods: params.authMethods,
-      pkpPublicKey: params.pkpPublicKey,
-      ...(params?.authSig && { authSig: params.authSig }),
-      // authSig: params.authSig,
-      siweMessage: siweMessageStr,
+      ...(params?.pkpPublicKey && { pkpPublicKey: params.pkpPublicKey }),
+      siweMessage: siweMessage,
+      curveType: LIT_CURVE.BLS,
+
+      // -- custom auths
+      ...(params?.litActionCode && { code: params.litActionCode }),
```

Comment:
Users can also pass ipfsId. Where's that param?

Reply:
Yea, they were missing - made the changes along with some test cases. https://github.com/LIT-Protocol/js-sdk/pull/453/files